presto-main/src/main/java/com/facebook/presto/operator/scalar/MergingDigest.java

```diff
@@ -767,155 +660,136 @@ public double quantile(double q) {
     }
 
     @Override
-    public Collection<Centroid> centroids() {
-        // we don't actually keep centroid structures around so we have to fake it
+    public Collection<Centroid> centroids()
+    {
         compress();
-        return new AbstractCollection<Centroid>() {
+        return new AbstractCollection<Centroid>()
+        {
             @Override
-            public Iterator<Centroid> iterator() {
-                return new Iterator<Centroid>() {
-                    int i = 0;
+            public Iterator<Centroid> iterator()
+            {
+                return new Iterator<Centroid>()
+                {
+                    int i;
 
                     @Override
-                    public boolean hasNext() {
-                        return i < lastUsedCell;
+                    public boolean hasNext()
+                    {
+                        return i < activeCentroids;
                     }
 
                     @Override
-                    public Centroid next() {
+                    public Centroid next()
+                    {
                         Centroid rc = new Centroid(mean[i], (int) weight[i], data != null ? data.get(i) : null);
                         i++;
                         return rc;
                     }
 
                     @Override
-                    public void remove() {
+                    public void remove()
+                    {
                         throw new UnsupportedOperationException("Default operation");
                     }
                 };
             }
 
             @Override
-            public int size() {
-                return lastUsedCell;
+            public int size()
+            {
+                return activeCentroids;
             }
         };
     }
 
     @Override
-    public double compression() {
+    public double compression()
+    {
         return publicCompression;
     }
 
     @Override
-    public int byteSize() {
+    public int byteSize()
+    {
         compress();
-        // format code, compression(float), buffer-size(int), temp-size(int), #centroids-1(int),
+        // format code, type code, min (double), max (double), compression(double), #centroids(int),
         // then two doubles per centroid
-        return lastUsedCell * 16 + 32;
-    }
-
-    @Override
-    public int smallByteSize() {
-        compress();
-        // format code(int), compression(float), buffer-size(short), temp-size(short), #centroids-1(short),
-        // then two floats per centroid
-        return lastUsedCell * 8 + 30;
+        return activeCentroids * 16 + 30;
     }
 
     @SuppressWarnings("WeakerAccess")
-    public ScaleFunction getScaleFunction() {
+    public ScaleFunction getScaleFunction()
+    {
         return scale;
     }
 
-    public enum Encoding {
-        VERBOSE_ENCODING(1), SMALL_ENCODING(2);
+    public enum Encoding
+    {
+        VERBOSE_ENCODING((byte) 0), SMALL_ENCODING((byte) 1);
```

Comment:
Are we not even supporting `SMALL_ENCODING`? If that's the case, just remove this, and remove encoding from serialization / deserialization.